components/DeleteButton.tsx

```diff
@@ -8,17 +10,35 @@ interface DeleteButtonProps {
 }
 
 export default function DeleteButton({ itemId, title }: DeleteButtonProps) {
-  // Get id of items you want to delete from database and refresh page
   if (typeof itemId === 'undefined') {
     throw new Error('item is undefined');
   }
-
   const router = useRouter();
+
+  const deleteItemHandler = async () => {
+    try {
+      const allConversations = await selectConversationsByItemId(itemId);
```

Comment:
I think I'm a little confused by this fetched being assigned to a variable called allConversations, as this is the name of our state where we care fetching and storing all our conversations. Since this is all conversations related to an item, maybe we could have something related to that? This just tripped me up a bit.

Reply:
Ah yeah, that's a good catch! `selectedConversations` ? 🤔